Problem in generating a new machine name with AWS IPI

When trying to add a new OCS worker node, the new node created successfully. 
But when searching for the old machine name at this line https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/node.py#L840 to know the new machine set, it didn't find it.
We probably need to change the method of searching the machine name.

Here is the failed test https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/187/testReport/tests.manage.z_cluster.nodes.test_node_replacement_proactive/TestNodeReplacement/test_nodereplacement_proactive/, and this is the Jenkins job: https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/187/. We also need to notice that we added in this test infra worker nodes.

@yitzhak12 , @prsurve ,
 I submitted a PR #4045 which fixes the pattern matching issue, but looks like another failure at the line https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/node.py#L848 where we are trying to label the node but it already exists, I guess because of machine set

```
E           ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: oc label node ip-10-0-220-113.us-east-2.compute.internal cluster.ocs.openshift.io/openshift-storage=''.
E           Error is error: 'cluster.ocs.openshift.io/openshift-storage' already has a value (), and --overwrite is false

ocs_ci/utility/utils.py:507: CommandFailed

```

is this the expected behavior ?

--- a/ocs_ci/ocs/node.py
+++ b/ocs_ci/ocs/node.py
@@ -844,10 +844,11 @@ def delete_and_create_osd_node_ipi(osd_node_name):
     log.info("Waiting for new worker node to be in ready state")
     machine.wait_for_new_node_to_be_ready(machineset_name)
     new_node_name = get_node_from_machine_name(new_machine_name)
-    log.info("Adding ocs label to newly created worker node")
-    node_obj = ocp.OCP(kind="node")
-    node_obj.add_label(resource_name=new_node_name, label=constants.OPERATOR_NODE_LABEL)
-    log.info(f"Successfully labeled {new_node_name} with OCS storage label")
+    if not is_node_labeled(new_node_name):
+        log.info("Adding ocs label to newly created worker node")
+        node_obj = ocp.OCP(kind="node")
+        node_obj.add_label(resource_name=new_node_name, label=constants.OPERATOR_NODE_LABEL)
+        log.info(f"Successfully labeled {new_node_name} with OCS storage label")
 
     return new_node_name
 

--- a/ocs_ci/ocs/node.py
+++ b/ocs_ci/ocs/node.py
@@ -847,7 +847,9 @@ def delete_and_create_osd_node_ipi(osd_node_name):
     if not is_node_labeled(new_node_name):
         log.info("Adding ocs label to newly created worker node")
         node_obj = ocp.OCP(kind="node")
-        node_obj.add_label(resource_name=new_node_name, label=constants.OPERATOR_NODE_LABEL)
+        node_obj.add_label(
+            resource_name=new_node_name, label=constants.OPERATOR_NODE_LABEL
+        )
         log.info(f"Successfully labeled {new_node_name} with OCS storage label")
 
     return new_node_name

--- a/ocs_ci/ocs/node.py
+++ b/ocs_ci/ocs/node.py
@@ -829,10 +829,11 @@ def delete_and_create_osd_node_ipi(osd_node_name):
     drain_nodes([osd_node_name])
     log.info("Getting machine name from specified node name")
     machine_name = machine.get_machine_from_node_name(osd_node_name)
+    machine_type = machine.get_machine_type(machine_name)
     log.info(f"Node {osd_node_name} associated machine is {machine_name}")
     log.info(f"Deleting machine {machine_name} and waiting for new machine to come up")
     machine.delete_machine_and_check_state_of_new_spinned_machine(machine_name)
-    new_machine_list = machine.get_machines()
+    new_machine_list = machine.get_machines(machine_type=machine_type)
     for machines in new_machine_list:
         # Trimming is done to get just machine name
         # eg:- machine_name:- prsurve-40-ocs-43-kbrvf-worker-us-east-2b-nlgkr

